model/src/cfd.rs

```diff
@@ -1403,10 +1379,45 @@ impl Cfd {
         }))
     }
 
+    fn event_with_reason(&self, event: EventKind, reason: anyhow::Error) -> CfdEvent {
+        self.log_cfd_event(&event, Some(reason));
+        self.issue_event_internal(event)
+    }
+
     fn event(&self, event: EventKind) -> CfdEvent {
+        self.log_cfd_event(&event, None);
+        self.issue_event_internal(event)
+    }
+
+    fn issue_event_internal(&self, event: EventKind) -> CfdEvent {
         CfdEvent::new(self.id, event)
     }
 
+    fn log_cfd_event(&self, event: &EventKind, reason: Option<anyhow::Error>) {
+        let message = match reason {
+            None => format!("Issuing event: {event}"),
+            Some(reason) => format!("Issuing event: {event} with reason: {reason:#}"),
+        };
+
+        match event {
+            EventKind::ContractSetupFailed
+            | EventKind::RolloverFailed
+            | EventKind::CollaborativeSettlementFailed => {
```

Comment:
🤔 You are silently promoting these two from warning to error. Is that what we want? I'm not necessarily against it, but the change shouldn't be buried in a refactoring commit.

Reply:
I added a separate commit for this

Reply:
I'm not sure we should promote these to errors right now. We append failures here when the other party rejected, and they're not error cases.

Reply:
What is an `error` for you, and how would you define when to use it?

I do agree that the `...Failed` events are strictly not critical errors if they don't happen repeatedly, but at the same time if we run into a failure when executing one of the protocols I feel we should get triggered to look into it. Maybe a warning is sufficient.
We treat `ContractSetupFailed` as `error` - would you downgrade that to be a warning then?

Reply:
I harmonized the cases to all be on `warn` now. I do think this is more correct, because strictly speaking non of these events happening is critical on it's own. 
At some point we said that `ContractSetup` is the one that should not fail, because it could lead to scenarios where one party publishes the `lock` transaction and the other one does not. Not sure if that is enough of a reason to log `ContractSetupFailed` on `error`. We generally might want to work on our alerts and configure to receive alerts on warnings - or a specific amount of warnings of a certain type within a time interval.